spec/datadog/ci/ext/environment_spec.rb

```diff
@@ -4,18 +4,58 @@
 require 'datadog/ci/ext/environment'
 
 RSpec.describe Datadog::CI::Ext::Environment do
+  fixture_dir = "#{File.dirname(__FILE__)}/fixtures/"
+
   describe '.tags' do
-    subject(:tags) { described_class.tags(env) }
+    subject(:tags) do
+      ClimateControl.modify(environment_variables) { described_class.tags(env) }
+    end
 
-    Dir.glob("#{File.dirname(__FILE__)}/fixtures/ci/*.json") do |filename|
+    let(:env) { {} }
+    let(:environment_variables) { {} }
+
+    shared_context 'without git installed' do
+      before { allow(described_class).to receive(:`).and_raise(Errno::ENOENT, 'git') }
+    end
```

Comment:
I'm... not entirely sure this is accurate. At least on my local machine:

```
[11] pry(main)> `this-command-does-not-exist`
Errno::ENOENT: No such file or directory - this-command-does-not-exist
from (pry):8:in ``'
[12] pry(main)> `this-command-does-not-exist 2>/dev/null`
=> ""
```

so while I think rather than `nil`, a bunch of the methods will return `""` if they fail, and we probably want to test with that.

Reply:
🙇 this is super clutch catch, I'll fix it.

Reply:
I always try to simplify system calls, but open3 is always the answer.